tee-worker/omni-executor/parentchain/signer/src/tx_signer.rs

```diff
@@ -86,28 +70,36 @@ impl<
 				transaction_version: runtime_version.transaction_version,
 			},
 		};
+		let nonce = self.nonce.fetch_add(1, Ordering::SeqCst);
+		if let Some(details) = call.validation_details() {
+			log::info!(
+				"Signing call {}::{} with nonce {}",
+				details.pallet_name,
+				details.call_name,
+				nonce
+			);
+		} else {
+			log::info!("Signing call with nonce {}", nonce);
+		}
 		let params = DefaultExtrinsicParamsBuilder::<ChainConfig>::new().nonce(nonce).build();
-		let signed_call = tx::create_signed(&call, &state, &signer, params).unwrap();
+		let signed_call = tx::create_signed(&call, &state, &self.signer, params).unwrap();
 
 		signed_call.encoded().to_vec()
 	}
 
-	pub fn get_signer_account_id(&self) -> AccountId {
-		self.get_signer().public_key().to_account_id().to_primitive_type()
+	fn get_signer_account_id(&self) -> AccountId {
+		self.signer.public_key().to_account_id().to_primitive_type()
 	}
 
-	fn get_signer(&self) -> Keypair {
-		let secret_key_bytes = self
-			.key_store
-			.read()
-			.map_err(|e| {
-				error!("Could not unseal key: {:?}", e);
-			})
-			.unwrap();
-		Keypair::from_secret_key(secret_key_bytes)
-			.map_err(|e| {
-				error!("Could not create secret key: {:?}", e);
-			})
-			.unwrap()
+	pub async fn update_nonce(&self) {
```

Comment:
If instance of `TxSigner` can be shared between threads and we have both in-memory nonce and method to reset it based on onchain nonce wouldn't it cause troubles when multiple threads shares it?

T1 sign (nonce 1)
T2 update (nonce 0)
T1 sign (nonce 1)

Reply:
We need a mechanism to get the current on-chain nonce if something wrong happens, otherwise the worker will never be able to send any transactions (if the nonce gets outdated for example). We could improve rpc client to add a retry mechanism and only update the nonce when needed